lib/Parse/ParseType.cpp

```diff
@@ -775,28 +776,26 @@ Parser::parseTypeIdentifier(bool isParsingQualifiedDeclBaseType) {
         Status.setHasCodeCompletionAndIsError();
         break;
       }
-      if (!peekToken().isContextualKeyword("Type")
-          && !peekToken().isContextualKeyword("Protocol")) {
-        // If parsing a qualified declaration name, break before parsing the
-        // period before the final declaration name component.
-        if (isParsingQualifiedDeclBaseType) {
-          // If qualified name base type cannot be parsed from the current
-          // point (i.e. the next type identifier is not followed by a '.'),
-          // then the next identifier is the final declaration name component.
-          BacktrackingScope backtrack(*this);
-          consumeStartingCharacterOfCurrentToken(tok::period);
-          if (!canParseBaseTypeForQualifiedDeclName())
-            break;
-        }
-        // Consume the period.
-        consumeToken();
-        continue;
+      if (peekToken().isContextualKeyword("Type") ||
+          peekToken().isContextualKeyword("Protocol"))
+        break;
```

Comment:
This doesn't seem right. In the old code, we would proceed to evaluate `Tok.is(tok::code_complete)` here, not break. I don't think your change preserves the behavior of the code, or if it does, it's not obvious to me.

Reply:
This is the same exact behavior. It is just that the break was not obvious in the old code at all, which is why this change may seem confusing, but this has the exact same behavior